# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/StorageAccess.cpp
# Commit: fa45bb12224b
# Full Hash: fa45bb12224bb79ffbb578dd9c3069357c830354
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2024-07-12 22:06:45
# Description:
#   Bug 1907353 - Use the loadInfo to check storage access if there is no target browsing context. r=anti-tracking-reviewers,pbz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D216394
# ==============================================================================

diff -r 9b522a0535b0 -r fa45bb12224b toolkit/components/antitracking/StorageAccess.cpp
--- a/toolkit/components/antitracking/StorageAccess.cpp	Fri Jul 12 13:59:09 2024 +0000
+++ b/toolkit/components/antitracking/StorageAccess.cpp	Fri Jul 12 13:59:13 2024 +0000
@@ -801,8 +801,7 @@
   bool isDocument = false;
   aChannel->GetIsDocument(&isDocument);
 
-  if (isDocument) {
-    MOZ_ASSERT(targetBC);
+  if (targetBC && isDocument) {
     nsCOMPtr<nsPIDOMWindowInner> inner =
         AntiTrackingUtils::GetInnerWindow(targetBC);
     if (inner && inner->UsingStorageAccess()) {
